src/plugins/embeddable/public/lib/embeddables/compatibility/legacy_embeddable_to_api.ts

```diff
@@ -135,6 +142,25 @@ export const legacyEmbeddableToApi = (
   const defaultPanelTitle = outputKeyToSubject<string>('defaultTitle');
   const disabledActionIds = inputKeyToSubject<string[] | undefined>('disabledActions');
 
+  function getSavedObjectId(input: { savedObjectId?: string }, output: { savedObjectId?: string }) {
+    return output.savedObjectId ?? input.savedObjectId;
+  }
+  const savedObjectId = new BehaviorSubject<string | undefined>(
+    getSavedObjectId(embeddable.getInput() as { savedObjectId?: string }, embeddable.getOutput())
```

Comment:
nit: You can get rid of this cast, and the other one on line 155 by changing `CommonLegacyInput` to `SavedObjectEmbeddableInput & { timeRange: TimeRange }`

Reply:
used `{ savedObjectId?: string }` instead of `SavedObjectEmbeddableInput` since `SavedObjectEmbeddableInput` requires `savedObjectId`